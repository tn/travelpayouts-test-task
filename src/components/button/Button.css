@import '../../Config.css';

.button {
  display: inline-block;
  margin: 0;
  padding: 10px 32px;
  font-family: var(--baseFontFamily);
  font-size: 19px;
  line-height: 28px;
  color: var(--defaultButtonTextColor);
  background-color: var(--defaultButtonColor);
  border: 1px solid var(--defaultButtonBorder);
  border-radius: 4px;
  cursor: pointer;
}

.button:hover,
.button:active {
  background-color: var(--defaultButtonHover);
}

.button--primary {
    padding: 12px 80px 10px;
    color: var(--primaryButtonTextColor);
    background-color: var(--primaryButtonColor);
    border: 0;
}

.button--primary:hover,
.button--primary:active {
    background-color: var(--primaryButtonHover);
}
