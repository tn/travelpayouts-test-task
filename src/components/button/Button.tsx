import * as React from 'react'
import './Button.css'

interface IButtonProps {
  primary?: boolean
  component?: any
  type?: string
  value?: string
}

const Button: React.SFC<IButtonProps> = ({ children, primary, component: Component = 'button', ...rest }) => (
  Component === 'input'
    ? <Component className={`button ${primary ? 'button--primary' : ''}`} {...rest} />
    : <Component className={`button ${primary ? 'button--primary' : ''}`} {...rest}>{children}</Component>
)

export default Button
