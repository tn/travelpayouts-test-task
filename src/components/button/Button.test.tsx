import * as React from 'react'
import * as Adapter from 'enzyme-adapter-react-16'
import { configure } from 'enzyme'
import * as renderer from 'react-test-renderer'
import Button from './Button'

configure({ adapter: new Adapter() })

describe('Button', () => {
  it('should be rendered', () => {
    const wrapper = renderer.create(<Button>Button</Button>).toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
