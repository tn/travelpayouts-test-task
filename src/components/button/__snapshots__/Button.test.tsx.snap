// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be rendered 1`] = `
<button
  className="button "
>
  Button
</button>
`;
